v2/blockstore/readonly.go

```diff
@@ -41,6 +41,15 @@ type ReadOnly struct {
 
 	// If we called carv2.NewReaderMmap, remember to close it too.
 	carv2Closer io.Closer
+
+	ropts carv2.ReadOptions
+}
+
+func UseWholeCIDs(enable bool) carv2.ReadOption {
```

Comment:
I put the blockstore-only options in the blockstore package, to help avoid confusion. The root carv2 APIs will simply ignore these. I don't think we can make them error there, because the options are forwarded when blockstore uses carv2.